waku/v2/node/storage/message/waku_message_store_queries.nim

```diff
@@ -76,7 +76,7 @@ proc createTable*(db: SqliteDatabase): DatabaseResult[void] {.inline.} =
 ## Create index
 
 template createIndexQuery(table: string): SqlQueryStr = 
-  "CREATE INDEX IF NOT EXISTS i_rt ON " & table & " (receiverTimestamp);"
+  "CREATE INDEX IF NOT EXISTS i_msg ON " & table & " (contentTopic, pubsubTopic, senderTimestamp, id);"
```

Comment:
Non-blocking:

> The queried node MUST sort the WakuMessages based on their Index, where the senderTime constitutes the most significant part and the digest comes next, and then perform pagination on the sorted result. As such, the retrieved page contains an ordered list of WakuMessages from the oldest message to the most recent one.
>
> – https://rfc.vac.dev/spec/13/

> `ReceiverTimestamp` is guaranteed to be set, while senders could omit setting `senderTimestamp`.
>
> – https://github.com/status-im/nwaku/blob/2cb7123df84b18ab7e4df0be0d77353cca4820f2/waku/v2/node/storage/message/waku_message_store.nim

1. Since the field is optional, where would messages like that end up in the resulting list? At the end?
2. In case of a null value, isn't worth falbackcing to `receiverTimestamp`, thus keeping it in the index?
3. Or shouldn't `senderTimestamp` be required?

Reply:
> 1. Since the field is optional, where would messages like that end up in the resulting list? At the end?

I am not sure about that. I need to review the current implementation.

> 2. In case of a null value, isn't worth fallbacking to _receiverTimestamp_, thus keeping it in the index?

What I am thinking is to simplify the current implementation by keeping a single column with a timestamp. 

The new column will be named `storeTimestamp` or `storedAt`. This timestamp will be set to _senderTimesamp_ if present and to the epoch time value at the moment of message insertion. The message will be dropped and not inserted into the store if the message is "too old" (older than the current time minus ~10s (?)) or "from the future" (current time plus more than ~10s (?)).

Optionally we can keep the _senderTimestamp_ column just in case we want to use it to extract metrics of the message propagation. But the reference timestamp column will be the `storeTimestamp` column for query purposes (and store restore functionality).

> 3. Or shouldn't _senderTimestamp_ be required?

IIRC For privacy-preserving reasons, the sender timestamp can be omitted. So makes no sense to make it mandatory.

@kaiserd @jm-clius  WDYT?